### PR TITLE
Add variable_with_name

### PR DESCRIPTION
**Pull request by blegat**
Resolve the first part of https://github.com/JuliaOpt/JuMP.jl/issues/1184

### CODE REVIEW

File: docs/src/variables.md [hunk 1]
```diff
@@ -246,6 +246,21 @@ julia> JuMP.fix_value(x)
 2.0
 ```
 
+## Variable names
+
+The name of a variable can retrieved using [`JuMP.name`](@ref), set by
```
Comment (mlubin) [inline docs/src/variables.md]: Clarify that this refers to the name attribute.

File: src/variables.jl [hunk 2]
```diff
@@ -223,6 +223,52 @@ function set_name(v::VariableRef, s::String)
     return MOI.set(owner_model(v), MOI.VariableName(), v, s)
 end
 
+"""
+    variable_with_name(model::AbstractModel, name::String)::AbstractVariableRef
```
Comment (mlubin) [inline src/variables.jl]: I'd suggest calling this `variable_by_name`.

File: src/variables.jl [hunk 3]
```diff
@@ -223,6 +223,52 @@ function set_name(v::VariableRef, s::String)
...
+
+Returns the reference of the variable with name `name`. Throws an error if
+several variables have the same name (even if it is different from `name`).
```
Comment (mlubin) [inline src/variables.jl]: > Throws an error if several variables have the same name

I thought we changed this behavior. It's only an error if multiple variables have the name that you're looking for, right?

File: src/variables.jl [hunk 4]
```diff
@@ -223,6 +223,52 @@ function set_name(v::VariableRef, s::String)
     return MOI.set(owner_model(v), MOI.VariableName(), v, s)
 end
 
+"""
+    variable_with_name(model::AbstractModel, name::String)::AbstractVariableRef
+
+Returns the reference of the variable with name `name`. Throws an error if
```
Comment (mlubin) [inline src/variables.jl]: What happens if the name doesn't exist? What error is thrown?

File: src/variables.jl (hunk 3 above)
Comment (blegat) [reply src/variables.jl]: Indeed!

File: src/variables.jl [hunk 5]
```diff
@@ -223,6 +223,73 @@ function set_name(v::VariableRef, s::String)
...
+ u[2]
+
+julia> JuMP.variable_by_name(model, "u[2]")
+u[2]
+```
+"""
+function variable_by_name(model::Model, name::String)
+    index = MOI.get(backend(model), MOI.VariableIndex, name)
+    if index isa Nothing
+        return nothing
```
[the remark this answers is not in the document]
Comment (blegat) [reply src/variables.jl]: This is more consistent with the behavior of search function in Base which returns `nothing` when the element is not present, e.g. `findfirst(isequal(3), [1, 2, 4])`.
The user will get an error later on if he uses the reference.

**Pull request merged**